contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -2653,6 +2653,10 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 					}
 					else if (isuser || isrole)
 					{
+						if (!has_privs_of_role(GetSessionUserId(), get_role_oid("sysadmin", false)))
+							ereport(ERROR,
+									(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
+									 errmsg("User does not have permission to perform this action.")));
```

Comment:
Here, we are missing following scenario.
```
create login l1;
grant sysadmin to l1;
-- connect using l1
create database db1;

revoke sysadmin from l1;
-- Now login l1 is still owns db1 even if it's not sysadmin, so if we login using l1 in db1 then we will get dbo user
-- connect using l1 in db1
select current_user() -> will be dbo

and since it is dbo, we will be allowed to drop role/users in current database db1.
```
Instead, We should verify that current user is dbo or not (OR member of db_owner or not).

Reply:
What you are saying is related to drop roles/users. 
This change only affects "create role" and "create user" statement.
Problem statement is to restrict T-SQL logins to create users and roles.

Reply:
Even creating role/users will be allowed in that case.

Reply:
I'm all good with the change apart from this, would you please add test for this and check if it matches expected behavior?

Reply:
There is an existing issue which will be tracked here. https://jira.rds.a2z.com/browse/BABEL-4652